core/src/main/scala/pureconfig/backend/ConfigFactoryWrapper.scala

```diff
@@ -22,15 +22,17 @@ object ConfigFactoryWrapper {
     unsafeToEither(ConfigFactory.load())
 
   /** @see `com.typesafe.config.ConfigFactory.parseFile()` */
-  def parseFile(path: Path): Either[ConfigReaderFailures, Config] =
-    unsafeToEither(ConfigFactory.parseFile(path.toFile))
+  def parseFile(path: Path): Either[ConfigReaderFailures, Config] = {
+    if (!path.toFile.isFile || !path.toFile.canRead) Left(ConfigReaderFailures(CannotReadFile(path)))
```

Comment:
Sure, I can change it to use NIO since we are already working with a `Path`. Out of curiosity, is there any functional difference in these checks between the two APIs?